server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -478,6 +501,75 @@ private RestResponse getDetailResponse(Cache<?, ?> cache) {
       return CompletableFuture.completedFuture(responseBuilder.status(OK).build());
    }
 
+   private CompletionStage<RestResponse> getCacheConfigMutableAttributes(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+
+      responseBuilder.contentType(APPLICATION_JSON);
+      if (!invocationHelper.getRestCacheManager().getInstance().getCacheConfigurationNames().contains(cacheName)) {
+         responseBuilder.status(NOT_FOUND).build();
+      }
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+      if (cache == null)
+         return notFoundResponseFuture();
+
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(cache.getAdvancedCache());
+      List<String> attributes = new ArrayList<>();
+      mutableAttributes(cacheConfiguration, attributes, null);
+      return asJsonResponseFuture(Json.make(attributes));
+   }
+
+   private static void mutableAttributes(ConfigurationElement<?> element, List<String> attributes, String prefix) {
+      prefix = prefix == null ? "" : element.name();
+      for (Attribute<?> attribute : element.attributes().attributes()) {
+         if (!attribute.isImmutable()) {
+            attributes.add(prefix + "." + attribute.getAttributeDefinition().name());
+         }
+      }
+      for (ConfigurationElement<?> child : element.children()) {
+         mutableAttributes(child, attributes, prefix);
+      }
+   }
+
+   private CompletionStage<RestResponse> getCacheConfigMutableAttribute(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String attributeName = request.getParameter("attribute-name");
+      responseBuilder.contentType(APPLICATION_JSON);
+      String cacheName = request.variables().get("cacheName");
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
+      if (cache == null) {
+         return notFoundResponseFuture();
+      }
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(cache.getAdvancedCache());
+      Attribute<?> attribute = cacheConfiguration.findAttribute(attributeName);
+      return asJsonResponseFuture(Json.make(String.valueOf(attribute.get())));
```

Comment:
I think we should return a `400` if the attribute is immutable.

Reply:
Done